package.json

```
@@ -179,7 +179,7 @@
     "release": "node infrastructure/tooling/src/main.js release",
     "staging-release": "node infrastructure/tooling/src/main.js staging-release",
     "release:publish": "node infrastructure/tooling/src/main.js release:publish",
-    "generate": "node infrastructure/tooling/src/main.js generate",
+    "generate": "export NODE_OPTIONS=--dns-result-order=ipv4first && node infrastructure/tooling/src/main.js generate",
```

Since Node.js v17, the DNS result order is returned how the resolver sent the list [PR](https://github.com/nodejs/node/pull/39987). This breaks some of our testing/infra scripts because IPv6 localhosts are now returned instead of IPv4.

---

Instead of interfering with DNS order returned by the operating system, it's probably better to make use of Happy Eyeballs*, which was finally implemented in Node v18:
```suggestion
    "generate": "export NODE_OPTIONS=--enable-network-family-autoselection && node infrastructure/tooling/src/main.js generate",
```
*Happy Eyeballs is used by every web browser. It also solves issues in round-robin cases where one returned IP address isn't accessible.

---

Awesome, thanks for this note @treysis!

---

@treysis it seems like this command line option is not available yet in v18:

```bash
/usr/local/bin/node: --enable-network-family-autoselection is not allowed in NODE_OPTIONS
```

---

I'll likely just revert to what we had and will come back to this if/when that CLI option is backported to v18.

@treysis do you happen to know if that's on the roadmap? I looked through some open issues/PRs and wasn't able to tell for sure.

---

Oops. No idea how I overlooked that. I was sure the cli option made it into 18.13. Apparently it's just the general support for Happy Eyeballs that made it into 18.13. It requires to set the option `autoSelectFamily=true` for each `net`-call separately.

Yes, it's a bit confusing. Seems to depend on https://github.com/nodejs/node/pull/45994 to make it into 18.x. It's worked on. So fingers crossed!